src/Datasource/Connection.php

```diff
@@ -33,17 +41,68 @@ class Connection
      */
     public function __construct(array $config)
     {
+        if (isset($config['name'])) {
+            $this->configName = $config['name'];
+        }
         $config = $this->_normalizeConfig($config);
-        /** @psalm-var class-string<\Muffin\Webservice\Webservice\Driver\AbstractDriver> */
         $driver = $config['driver'];
         unset($config['driver'], $config['service']);
 
-        $this->_driver = new $driver($config);
+        $tempDriver = new $driver($config);
 
-        /** @psalm-suppress TypeDoesNotContainType */
-        if (!($this->_driver instanceof AbstractDriver)) {
+        if (!($tempDriver instanceof AbstractDriver)) {
```

Comment:
The value from $config['driver'] on line 48 is not guaranteed to be AbstractDriver, it is mixed, so it seems to me that this check is required, especially because the property is typed.

Reply:
If a wrong type is assigned to the property then PHP itself will throw a TypeError, so we don't need to explicitly throw one.

Reply:
You can use `asset()` instead to make phpstan.

Reply:
Implemented